PROVESFlightControllerReference/Components/TcSecurityDeframer/Authenticator.cpp

```
@@ -0,0 +1,121 @@
+
+    // Set up the key attributes
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_set_key_type(&attributes, PSA_KEY_TYPE_HMAC);
+    psa_set_key_bits(&attributes, static_cast<size_t>(Ccsds355_0_B_2::kTCSecurityTrailer * 8));
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_VERIFY_MESSAGE);
+    psa_set_key_algorithm(&attributes, PSA_ALG_AT_LEAST_THIS_LENGTH_MAC(PSA_ALG_HMAC(PSA_ALG_SHA_256),
+                                                                        Ccsds355_0_B_2::kTCSecurityTrailer));
+    psa_set_key_lifetime(&attributes, PSA_KEY_LIFETIME_VOLATILE);
+
+    // Import the key into PSA key store
+    const psa_status_t status = psa_import_key(&attributes, keyBytes, sizeof(keyBytes), &keyId);
```

Clang (14.0.6)

[warning] variable 'status' is not initialized

(cppcoreguidelines-init-variables)